File: 541-reverse-string-ii/reverse-string-ii.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        slow = 0
        fast = k
        stack = []
        arr = []
        while fast <= len(s) or slow < len(s):
            while slow < fast and slow < len(s):
                stack.append(s[slow])
                slow += 1
            slow += k
            while stack:
                arr.append(stack.pop())
            while fast < slow and fast < len(s):
                arr.append(s[fast])
                fast += 1
            fast += k
        return ''.join(arr)

File: 541-reverse-string-ii/reverse-string-ii.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var reverseStr = function (s, k) {
    let slow = 0
    let fast = k
    let stack = []
    let arr = []
    while (fast <= s.length || slow < s.length) {
        while (slow < fast) {
            stack.push(s[slow])
            slow++
        }
        slow += k
        while (stack.length > 0) {
            arr.push(stack.pop())

        }
        while (fast < slow) {
            arr.push(s[fast])
            fast++
        }
        fast += k
    }
    return arr.join("")
};